core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -161,6 +160,20 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     new HistoryServerDiskManager(conf, path, listing, clock)
   }
 
+  private val processing = new ConcurrentSkipListSet[String]
```

Comment:
How about using `ConcurrentHashMap` instead of `ConcurrentSkipListSet`.   
[`ConcurrentSkipListSet`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentSkipListSet.html) takes more cost (i.e. `O(log(n))`) compared to `O(1)` in `ConcurrentHashMap`.

Reply:
thanks, done.